src/main/java/racingcar/model/Car.java

```diff
@@ -0,0 +1,48 @@
+package racingcar.model;
+
+import java.util.Objects;
+
+public class Car {
+    private final Name name;
+    private Position position;
+
+    public Car(final Name name) {
+        this.name = name;
+        this.position = new Position();
+    }
+
+    public void move(boolean isAbleToMove) {
+        if (isAbleToMove) {
+            position = position.move();
+        }
+    }
+
+    public boolean hasBiggerPositionThan(Car car) {
+        return position.isBiggerOrEqualTo(car.position);
+    }
+
+    public Position getPosition() {
+        return position;
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (!(object instanceof Car)) {
+            return false;
+        }
+        Car car = (Car) object;
+        return Objects.equals(name, car.name);
+    }
```

Comment:
```suggestion
    @Override
    public boolean equals(Object object) {
        if (this == object) {
            return true;
        }
        if (!(object instanceof Car car)) {
            return false;
        }
        return Objects.equals(name, car.name);
    }
```
요 부분 패턴매칭 사용하는 것은 어떻게 생각하세요? 패턴 매칭을 사용하면 타입 체크와 동시에 변수 바인딩이 이루어져서 바로 사용할 수 있어요.

Reply:
오 이런 방식으로 변수를 초기화할 수도 있군요!